src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java

```diff
@@ -308,22 +310,38 @@ private void handleInvalidFileError(
             null
         );
 
-        ErrorMapping
-            .getFor(cause.getClass())
-            .ifPresent(errorCode -> {
-                try {
-                    sendErrorMessageToQueue(zipFilename, containerName, eventId, errorCode, cause);
-                } catch (Exception exc) {
-                    log.error(
-                        "Error sending notification to the queue."
-                            + "File name: " + zipFilename + " "
-                            + "Container: " + containerName,
-                        exc
-                    );
-                }
-            });
+        Optionals.ifPresentOrElse(
+            ErrorMapping.getFor(cause.getClass()),
+            (errorCode) -> trySendErrorMessageAndMoveFileToRejectedContainer(
+                zipFilename,
+                containerName,
+                cause, eventId,
+                leaseId,
+                errorCode
+            ),
+            () -> {
+                throw new ConfigurationException("Error code mapping not found for " + cause.getClass().getName());
+            }
+        );
+    }
 
-        blobManager.tryMoveFileToRejectedContainer(zipFilename, containerName, leaseId);
+    private void trySendErrorMessageAndMoveFileToRejectedContainer(
+        String zipFilename,
+        String containerName,
+        Exception cause,
+        Long eventId,
+        String leaseId,
+        ErrorCode errorCode
+    ) {
+        try {
+            sendErrorMessageToQueue(zipFilename, containerName, eventId, errorCode, cause);
+            blobManager.tryMoveFileToRejectedContainer(zipFilename, containerName, leaseId);
+        } catch (Exception exc) {
+            final String msg = "Error sending notification to the queue."
+                + "File name: " + zipFilename + " "
+                + "Container: " + containerName;
```

Comment:
The message is about sending notification to the queue, but the try part is about more than that. Perhaps we should remove the call to blobManager from this method.

Reply:
Done